apps/wallet/src/ui/app/redux/slices/originbyte-nfts/NftClient.ts

```diff
@@ -0,0 +1,269 @@
+
+type NftDomains = {
+    url: string;
+    name: string;
+    description: string;
+};
+
+const isTypeMatchRegex = (d: GetObjectDataResponse, regex: RegExp) => {
+    const { details } = d;
+    if (isSuiObject(details)) {
+        const { data } = details;
+        if (isSuiMoveObject(data)) {
+            return data.type.match(regex);
+        }
+    }
+    return false;
+};
+
+export const parseDomains = (domains: GetObjectDataResponse[]) => {
+    const response: Partial<NftDomains> = {};
+    const urlDomain = domains.find((d) => isTypeMatchRegex(d, UrlDomainRegex));
+    const displayDomain = domains.find((d) =>
+        isTypeMatchRegex(d, DisplayDomainRegex)
+    );
+
+    if (
+        urlDomain &&
+        isSuiObject(urlDomain.details) &&
+        isSuiMoveObject(urlDomain.details.data)
+    ) {
+        const { data } = urlDomain.details;
+        response.url = (data.fields as UrlDomainRpcResponse).value.fields.url;
+    }
+    if (
+        displayDomain &&
+        isSuiObject(displayDomain.details) &&
+        isSuiMoveObject(displayDomain.details.data)
+    ) {
+        const { data } = displayDomain.details;
+        response.description = (
+            data.fields as DisplayDomainRpcResponse
+        ).value.fields.description;
+        response.name = (
+            data.fields as DisplayDomainRpcResponse
+        ).value.fields.name;
+    }
+
+    return response;
+};
+
+export class NftClient {
+    constructor(private provider: JsonRpcProvider) {}
+
+    private fetchObjectIdsForAddress = async <RpcResponse, DataModel>(
+        address: string,
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ): Promise<string[]> => {
+        const objectsForWallet = await this.provider.getObjectsOwnedByAddress(
+            address
+        );
+        return objectsForWallet
+            .filter((_) => _.type.match(parser.regex))
+            .map((_) => _.objectId);
+    };
+
+    parseObjects = async <RpcResponse, DataModel>(
+        objects: GetObjectDataResponse[],
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ): Promise<DataModel[]> => {
+        const parsedObjects = objects
+            .filter(isObjectExists)
+            .map((_) => {
+                if (
+                    isSuiObject(_.details) &&
+                    isSuiMoveObject(_.details.data) &&
+                    _.details.data.type.match(parser.regex)
+                ) {
+                    return parser.parser(
+                        _.details.data.fields as RpcResponse,
+                        _.details,
+                        _
+                    );
+                }
+                return undefined;
+            })
+            .filter((_): _ is DataModel => !!_);
+
+        return parsedObjects;
+    };
+
+    fetchAndParseObjectsById = async <RpcResponse, DataModel>(
+        ids: string[],
+        parser: SuiObjectParser<RpcResponse, DataModel>
```

Comment:
Why bother with this dynamic parsing logic if it's not actually changed out ever?